subprojects/docs/src/samples/userguide/artifacts/dependencyBasics/build.gradle

```diff
@@ -1,10 +1,10 @@
-apply plugin: 'java'
+apply plugin: 'java-library'
 
 repositories {
     mavenCentral()
 }
 
 dependencies {
-    compile group: 'org.hibernate', name: 'hibernate-core', version: '3.6.7.Final'
-    testCompile group: 'junit', name: 'junit', version: '4.+'
+    implementation 'org.hibernate:hibernate-core:3.6.7.Final'
+    testImplementation 'junit:junit:4.+'
```

Comment:
There is a section which explains configurations in more detail - which is good. But the three configurations users should know about in the beginning, in my opinion, are `implementation`, `api`, `testImplementation`. I think that's what we want to establish as the "new way" of declaring dependencies with the Java plugins.

Thus I would propose to add `api` to the example. We could use guava and add that after the sentence about hibernate. Something like:
Guava is also required to compile the project and is also part of the project's api as it provided types such as `Multimap` that we want to use in public interfaces.

Reply:
That's a great suggestion. I will work that in right away.

Reply:
Having the list of these three configurations is a great addition. Thanks! I was also thinking about adding one sentence about Guava when the example is explained. After
_The Project.dependencies{} code block declares that Hibernate core 3.6.7.Final is required to compile the project’s production source code._

Bit maybe we do not need that with this new explanation further down.